substrate/dex/pallet/src/lib.rs

```diff
@@ -362,35 +408,57 @@ pub mod pallet {
           } else {
             0
           };
-        let sri_per_coin = sri_per_coin.to_be_bytes();
+        let sri_per_coin_bytes = sri_per_coin.to_be_bytes();
+
+        SpotPriceForBlock::<T>::set(n, coin, sri_per_coin_bytes);
 
-        SpotPriceForBlock::<T>::set(n, coin, sri_per_coin);
+        // Include this spot price into low-to-high multiset
+        {
+          let observed = OraclePrices::<T>::get(coin, sri_per_coin_bytes).unwrap_or(0);
+          OraclePrices::<T>::set(coin, sri_per_coin_bytes, Some(observed + 1));
+        }
 
-        // Include this spot price into the multiset
+        // into high-to-low multiset
         {
-          let observed = OraclePrices::<T>::get(coin, sri_per_coin).unwrap_or(0);
-          OraclePrices::<T>::set(coin, sri_per_coin, Some(observed + 1));
+          let reverse = reverse_lexicographic_order(sri_per_coin_bytes);
+          let observed = OraclePricesReverse::<T>::get(coin, reverse).unwrap_or(0);
+          OraclePricesReverse::<T>::set(coin, reverse, Some(observed + 1));
         }
 
         // pop the earliest key from the window once we reach its full size.
-        if n >= ORACLE_WINDOW_SIZE.into() {
-          let start_of_window = n - ORACLE_WINDOW_SIZE.into();
+        let removed = if n >= MEDIAN_PRICE_WINDOW_SIZE.into() {
+          let start_of_window = n - MEDIAN_PRICE_WINDOW_SIZE.into();
           let start_spot_price = Self::spot_price_for_block(start_of_window, coin);
           SpotPriceForBlock::<T>::remove(start_of_window, coin);
-          // Remove this price from the multiset
+          // Remove this price from the low-to-high multiset
           OraclePrices::<T>::mutate_exists(coin, start_spot_price, |v| {
             *v = Some(v.unwrap_or(1) - 1);
             if *v == Some(0) {
               *v = None;
             }
           });
-        }
+
+          // Remove this price from the high-to-low multiset
+          let reverse = reverse_lexicographic_order(start_spot_price);
+          OraclePricesReverse::<T>::mutate_exists(coin, reverse, |v| {
+            *v = Some(v.unwrap_or(1) - 1);
```

Comment:
Shouldn't this just be an unwrap? There's no way for it to not be here if it's being removed.